tgui/packages/tgui/interfaces/ApcControl.js

```diff
@@ -0,0 +1,335 @@
+          checked={sortByField === 'name'}
+          content="Name"
+          onClick={() => setSortByField(sortByField !== 'name' && 'name')} />
+        <Button.Checkbox
+          checked={sortByField === 'charge'}
+          content="Charge"
+          onClick={() => setSortByField(
+            sortByField !== 'charge' && 'charge'
+          )} />
+        <Button.Checkbox
+          checked={sortByField === 'draw'}
+          content="Draw"
+          onClick={() => setSortByField(sortByField !== 'draw' && 'draw')} />
+        {emagged === 1 && (
+          <Fragment>
+            <Button color={(logging === 1 ? "bad" : "good")}
+              content={(logging === 1 ? "Stop Logging" : "Restore Logging")}
+              onClick={() => act('toggle-logs')}
+              right={13}
+              position="fixed"
+            />
+            <Button content="Reset Console"
+              onClick={() => act('restore-console')}
+              right={5}
+              position="fixed"
+            />
+          </Fragment>
+        )}
+        <Button color={'bad'}
+          content="Log Out"
+          right={0}
+          position="fixed"
+          tooltip={'Log Out'}
+          onClick={() => act('log-out')}
+        />
+      </Box>
+      <Table.Row header>
+        <Table.Cell>
+          Area
+        </Table.Cell>
+        <Table.Cell collapsing>
+          Charge
+        </Table.Cell>
+        <Table.Cell collapsing textAlign="right">
+          Draw
+        </Table.Cell>
+        <Table.Cell collapsing title="Equipment">
+          Eqp
+        </Table.Cell>
+        <Table.Cell collapsing title="Lighting">
+          Lgt
+        </Table.Cell>
+        <Table.Cell collapsing title="Environment">
+          Env
+        </Table.Cell>
+      </Table.Row>
+    </Section>
+  );
+};
+
+const ApcControlScene = (props, context) => {
+  const { data, act } = useBackend(context);
+
+  const [
+    sortByField,
+  ] = useLocalState(context, 'sortByField', null);
+
+  const apcs = flow([
+    map((apc, i) => ({
+      ...apc,
+      // Generate a unique id
+      id: apc.name + i,
+    })),
+    sortByField === 'name' && sortBy(apc => apc.name),
+    sortByField === 'charge' && sortBy(apc => -apc.charge),
+    sortByField === 'draw' && sortBy(
+      apc => -powerRank(apc.load),
+      apc => -parseFloat(apc.load)),
+  ])(data.apcs);
+  return (
+    <Table>
+      {apcs.map((apc, i) => (
+        <Table.Row header
+          key={apc.id}
+          className="Table__row candystripe">
+          <Table.Cell>
+            <Button
+              icon={apc.operating ? 'power-off' : 'times'}
+              color={apc.operating ? 'good' : 'bad'}
+              onClick={() => act('breaker',
+                { ref: apc.ref }
+              )}
+            />
+          </Table.Cell>
+          <Table.Cell>
+            <Button
+              onClick={() => act('access-apc',
+                { ref: apc.ref }
+              )}>
```

Comment:
Write it this way please.

```suggestion
              onClick={() => act('access-apc', {
                ref: apc.ref,
              })}>
```

Reply:
Alright